wp: add page (#5985)
tldr-pages/tldr: pages/common/wp.md

# wp

> The official command-line interface to manage WordPress instances.
> More information: <https://wp-cli.org/>.

> - Describe the current system environment:

Reviewer: I have never used ws.
what is a current system environment, will a ws beginner know what that is?

Author: Someone familiar with WordPress might be able to assume, but it could more explicit if preferred.
It'll display the operating system, and paths/versions of the shell, PHP installation, and WP-CLI installation.

It displays a lot of information, so I'm not sure how to concisely describe it all without being too vague.

> ```sh
> $ wp --info
> OS:	Darwin 16.7.0 Darwin Kernel Version 16.7.0: Thu Jan 11 22:59:40 PST 2018; root:xnu-3789.73.8~1/RELEASE_X86_64 x86_64
> Shell:	/bin/zsh
> PHP binary:    /usr/local/bin/php
> PHP version:    7.0.22
> php.ini used:   /etc/local/etc/php/7.0/php.ini
> WP-CLI root dir:        /home/wp-cli/.wp-cli/vendor/wp-cli/wp-cli
> WP-CLI vendor dir:	    /home/wp-cli/.wp-cli/vendor
> WP-CLI packages dir:    /home/wp-cli/.wp-cli/packages/
> WP-CLI global config:   /home/wp-cli/.wp-cli/config.yml
> WP-CLI project config:
> WP-CLI version: 2.4.0
> ```
> \- https://wp-cli.org/#installing

Might something like this be better? 🤔 

```suggestion
- Describe the operating system, shell, PHP, and WP-CLI installation:
```

Reviewer: Thats better IMO

Reviewer 2: I like that better too 👍🏻

> - Describe the operating system, shell, PHP, and WP-CLI installation:

Reviewer: ```suggestion
- Describe the operating system, shell, PHP and WP-CLI installation:
```

Small error I missed

Author: Hmm, that's not an error, that's an [Oxford comma](https://en.wikipedia.org/wiki/Serial_comma).

There are many examples of it throughout this repository.

![image](https://user-images.githubusercontent.com/22801583/118687128-f4b7d380-b804-11eb-905c-11b0f304f6a1.png)

Reviewer 2: Ah, interesting! I think we can keep it.

By the way @SethFalco did you know that in regex `\w` = `[a-zA-Z0-9$_]`? Using `\w` would shorten that regex.

Author: Ahh, nice! Thanks for the heads up. \\(^-^)/

> - Describe the operating system, shell, PHP, and WP-CLI installation:

Reviewer 2: Add an appositive to clarify? (assuming `WP-CLI` _is_ `ws`)
```suggestion
- Describe the operating system, shell, PHP, and WP-CLI (`ws`) installation:
```

Author: Have I done something wrong somewhere?

Where is `ws` coming from? ^-^' @CleanMachine1 also mentioned `ws`, but I assumed it was just a typo of `wp`.

Reviewer: it was

Reviewer 2: Oops, I think I was getting it from @CleanMachine1's comment 😄

Suggestion updated.

> - Describe the operating system, shell, PHP, and WP-CLI (`wp`) installation:

Reviewer 3: Does this command print whole sentences? If not I think 
```suggestion
- Print information about the operating system, shell, PHP, and WP-CLI (`wp`) installation:
```
would be more fitting.

`wp --info`

- Update WP-CLI:

`wp cli update`

- Install and activate a WordPress plugin:

`wp plugin install {{plugin}} --activate`

> - Replace all instances of the old string in the database with the new string:
> 
> `wp search-replace {{old}} {{new}}`

Reviewer 3: ```suggestion
- Replace all instances of a string in the database:

`wp search-replace {{old_string}} {{new_string}}`
```
I think "replace" already implies that it is replaced with something new. Also, I think more descriptive parameters would be helpful.

Author: I agree with this one, I did the original mostly because in the WordPress documentation that's how they described it. I figured maybe it was preferred for broader audiences. 🤔 

> Searches through all rows in a selection of tables and replaces appearances of the first string with the second string.
> \- https://developer.wordpress.org/cli/commands/search-replace/

I agree with the improvement, just wanted to note here why I did the original wording.
